src/app/seller-post-ad-form/seller-post-ad-form.component.html

```diff
@@ -0,0 +1,129 @@
+<div id="post-ad-container">
+    <form #postAdForm="ngForm" (ngSubmit)="onPost(postAdForm)">
```

Comment:
Lets discuss which is better way for form binding. 
By this way, component is not aware of model used by form in advance, by this way.
You should design/write this component in such a way that same can be used for update data.
E.g.  product is null then its post call ( new ad ) and if product has some value then its update call ( update ad)

Html is driving model which seems wrong.